packages/x-date-pickers/src/DateCalendar/tests/DateCalendar.test.tsx

```diff
@@ -68,17 +65,17 @@ describe('<DateCalendar />', () => {
       />,
     );
 
-    fireEvent.click(screen.getByText('January 2019'));
+    await user.click(screen.getByText('January 2019'));
     expect(screen.queryByText('January 2019')).toBeVisible();
     expect(screen.queryByLabelText('year view is open, switch to calendar view')).to.equal(null);
 
-    fireEvent.click(screen.getByTitle('Previous month'));
+    await user.setup({ pointerEventsCheck: 0 }).click(screen.getByTitle('Previous month'));
     expect(onMonthChangeMock.callCount).to.equal(0);
 
-    fireEvent.click(screen.getByTitle('Next month'));
+    await user.setup({ pointerEventsCheck: 0 }).click(screen.getByTitle('Next month'));
     expect(onMonthChangeMock.callCount).to.equal(0);
 
-    fireEvent.click(screen.getByRole('gridcell', { name: '5' }));
+    await user.setup({ pointerEventsCheck: 0 }).click(screen.getByRole('gridcell', { name: '5' }));
```

Comment:
The picker is `disabled`, so in the browser it complains that you can't perform click on disabled elements. We ignore the disabled check.